src/Native/System.Native/pal_mount.cpp

```diff
@@ -0,0 +1,144 @@
+#else
+#include <sys/statfs.h>
+#include <mntent.h>
+#define STRING_BUFFER_SIZE 8192
+#endif
+
+#if HAVE_MNTINFO
+
+int GetMountInfo(MountPointFound onFound)
+{
+    // getmntinfo returns pointers to OS-internal structs, so we don't need to worry about free'ing the object
+    int result = 0;
+    struct statfs* mounts = nullptr;
+    int count = getmntinfo(&mounts, 0);
+    if (count > 0)
+    {
+        for (int32_t i = 0; i < count; i++)
+        {
+            int32_t nameLength = strlen(mounts[i].f_mntonname);
+            onFound(mounts[i].f_mntonname, nameLength);
+        }
+    }
+    else
+    {
+        // Something went wrong so set our result to errno
+        result = errno;
+    }
+    
+    return result;
+}
+
+#else
+
+int GetMountInfo(MountPointFound onFound)
+{
+    int result = 0;
+    FILE* fp = setmntent("/proc/mounts", MNTOPT_RO);
+    if (fp == nullptr)
+    {
+        result = errno;
+    }
+    else
+    {
+        // The _r version of getmntent needs all buffers to be passed in; however, we don't know how big of a string
+        // buffer we will need, so pick something that seems like it will be big enough.
+        char buffer[STRING_BUFFER_SIZE] = { };
+        mntent entry = { };
+        while (getmntent_r(fp, &entry, buffer, STRING_BUFFER_SIZE) != nullptr)
+        {
+            int32_t nameLength = strlen(entry.mnt_dir);
+            onFound(entry.mnt_dir, nameLength);
+            memset(buffer, 0, STRING_BUFFER_SIZE);
+        }
+
+        assert(endmntent(fp) == 1); // documented to always return 1
+        result = 0; // We need to standardize a success return code between our implementations, so settle on 0 for success
+    }
+
+    return result;
+}
+
+#endif
+
+extern "C"
+int GetAllMountPoints(MountPointFound onFound)
+{
+    return GetMountInfo(onFound);
+}
+
+extern "C"
+int GetSpaceInfoForMountPoint(
+    const char*             name,
+    MountPointInformation*  mpi)
+{
+    assert(name != nullptr);
+    assert(mpi != nullptr);
+
+    struct statfs stats = { };
+    int result = statfs(name, &stats);
+    if (result == 0)
+    {
+        mpi->AvailableFreeSpace = (int64_t)(stats.f_bsize * stats.f_bavail);
+        mpi->TotalFreeSpace = (int64_t)(stats.f_bsize * stats.f_bfree);
+        mpi->TotalSize = (int64_t)(stats.f_bsize * stats.f_blocks);
+    }
+    else
+    {
+        result = errno;
+    }
+
+    return result;
+}
+
+extern "C"
+int GetFormatInfoForMountPoint(
+    const char* name, 
+    char*       formatNameBuffer, 
+    int32_t     bufferLength, 
+    int64_t*    formatType)
+{
+    assert((formatNameBuffer != nullptr) && (formatType != nullptr));
+    assert(bufferLength > 0);
+
+    struct statfs stats = { };
+    int result = statfs(name, &stats);
+    if (result == 0)
+    {
+
+#if HAVE_STATFS_FSTYPENAME
+        if (bufferLength < MFSNAMELEN)
+        {
+            result = ERANGE;
+        }
+        else
+        {
+            strcpy(formatNameBuffer, stats.f_fstypename);
```

Comment:
Good catch; I added a pal_utilities.cpp file that will abstract away the safe string copy function into a platform specific one that can be used elsewhere in the PAL. This is necessary since different platforms have different safe string copy functions (ie Ubuntu has strcpy_s but OS X has strlcpy and they have different parameters).